frontend/src/hooks/query/use-settings.ts

```diff
@@ -5,6 +5,16 @@ import OpenHands from "#/api/open-hands";
 import { useAuth } from "#/context/auth-context";
 import { DEFAULT_SETTINGS } from "#/services/settings";
 
+// Instead of directly using useLocation, we'll check the current path manually
+// This avoids the Router context requirement
+const isOnTosPage = () => {
```

Comment:
this duplicated function is suspicious. I'm sure there's a better way to do this. It seems like just a modal so it should be able to pop up anywhere...